src/collapse/mod.rs

```diff
@@ -94,316 +83,27 @@ pub trait Collapse {
     /// - `Some(false)` means "no, this implementation definitely won't work"
     fn is_applicable(&mut self, input: &str) -> Option<bool>;
 
-    #[cfg(test)]
-    fn set_nstacks_per_job(&mut self, n: usize);
-
-    // Temporarily public for tests.
-    // TODO: Restructure tests so that we don't expose this.
-    #[doc(hidden)]
+    /// Sets the number of threads to use. Default is the number of logical cores on your machine.
     fn set_nthreads(&mut self, n: usize);
```

Comment:
Is there a reason you decided to make this part of the `Collapse` trait? I don't think implementations of this trait should be required to be multi-threaded. Besides, it seems strange to have two different ways to set the number of threads: with this method, and with the `Options` of each folder.

Reply:
I think you did that, not me? :p

Reply:
Not me. Did you think you were replying to @bcmyers?

Reply:
Oh, yes, sorry!

Reply:
@jsonrhansen - Good point. I'll switch back sample to run on just one thread, per your other comments, and take this method off the `Collapse` trait.